dali/operators/reader/loader/video/frames_decoder_test.cc

```diff
@@ -184,6 +194,18 @@ TEST_F(FramesDecoderTest_CpuOnlyTests, InvalidSeek) {
   }
 }
 
+TEST_F(FramesDecoderTest_CpuOnlyTests, InvalidCodec) {
+  std::string path = testing::dali_extra_path() + "/db/video/vp9/vp9_0.mp4";
+
+  try {
+    FramesDecoder file(path);
+  } catch (const DALIException &e) {
+    EXPECT_TRUE(
+      strstr(e.what(),
+      make_string("Unsupported video codec: vp9 in file: ", path).c_str()));
+  }
```

Comment:
This test will pass if no exception is thrown, right? Do we want to fail in such case?

Reply:
Nicely caught. I extracted some helper functions to run these testes where we expect `FramesDecoder` to throw and refactored all of those tests that do that. In the helper I added a `FAIL()`, if we do not throw.